nannies/manila-missing-snapshot.py

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env python
+#
+# Copyright (c) 2022 SAP SE
+# All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+#
+import yaml
+
+from helper.manilananny import ManilaNanny, base_command_parser
+from helper.prometheus_exporter import LabelGauge
+
+
+class MissingSnapshotNanny(ManilaNanny):
+    def __init__(self, config_file, netapp_filers, interval, prom_port) -> None:
+        super(MissingSnapshotNanny, self).__init__(config_file, interval, prom_port)
+        with open(netapp_filers, "r") as f:
+            self.netapp_filers = yaml.safe_load(f)
+
+        # intialize gauge
+        self.missing_snapshot_gauge = LabelGauge(
+            "manila_nanny_missing_snapshot_instance",
+            "snapshot not found on netapp filer",
+            ["id", "snapshot_id", "status"],
+        )
+
+    def _run(self):
+        print("export missing snapshots")
+        self.missing_snapshot_gauge.export(self._get_missing_snapshots())
+
+    def _get_missing_snapshots(self):
+        """get list of snapshots that are not found on netapp filer"""
+        # list snapshots from Manila; and build a hash table with snapshot id as key
+        manilaclient = self.get_manilaclient("2.19")
+        search_opts = {"all_tenants": True}
+        resp = manilaclient.share_snapshot_instances.list(search_opts=search_opts)
+        snapshot_table = {s._info["id"]: s._info for s in resp}
```

Comment:
Can you add a comment with an example how `s._info` looks like, please? Looking at this I want to know which keys will be present.

Reply:
yes, attributes can be used instead of accessing internal variable https://github.com/sapcc/openstack-nannies/pull/56/commits/703c1fbb8ac3048e515e4f6890459df017754bb4